src/Pretzel.Logic/Templating/JekyllEngineBase.cs

```diff
@@ -162,6 +267,15 @@ private void ProcessFile(string outputDirectory, Page page, Page previous, Page
                     : Context.ExcerptSeparator;
                 try
                 {
+                    if ((context.Paginator != null) && (context.Paginator.Posts != null))
+                    {
+                        foreach (var post in context.Paginator.Posts)
```

Comment:
Is it really necessary to re-render and get excerpt for all posts since we already doing this just a few lines after?